infra/aws/terraform/modules/eks-prow-iam/eks_resources_boundary.tf

```diff
@@ -14,20 +14,21 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-resource "aws_iam_policy" "provisioner_permission_boundary" {
-  name        = "ProvisionerPermissionBoundary"
+resource "aws_iam_policy" "eks_resources_permission_boundary" {
```

Comment:
Same as earlier, let's add a comment describing this boundary (i.e. what we have in the README).